src/Domain/CoordinateMaps/Wedge3D.cpp

```diff
@@ -215,79 +148,46 @@ tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame> Wedge3D::jacobian(
       make_with_value<tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame>>(
           dereference_wrapper(source_coords[0]), 0.0);
 
-  auto &dx_dxi = get<0, 0>(jacobian_matrix),
-       &dx_deta = get<0, 1>(jacobian_matrix),
-       &dx_dzeta = get<0, 2>(jacobian_matrix),
-       &dy_dxi = get<1, 0>(jacobian_matrix),
-       &dy_deta = get<1, 1>(jacobian_matrix),
-       &dy_dzeta = get<1, 2>(jacobian_matrix),
-       &dz_dxi = get<2, 0>(jacobian_matrix),
-       &dz_deta = get<2, 1>(jacobian_matrix),
-       &dz_dzeta = get<2, 2>(jacobian_matrix);
-
-  dz_dxi = -second_blending_factor_over_rho_cubed * cap_xi * cap_xi_deriv;
-  dz_deta = -second_blending_factor_over_rho_cubed * cap_eta * cap_eta_deriv;
-  dz_dzeta = first_blending_rate + second_blending_rate_over_rho;
-  dx_dxi = (first_blending_factor +
-            second_blending_factor_over_rho_cubed * (1.0 + square(cap_eta))) *
-           cap_xi_deriv;
-  dx_deta = dz_deta * cap_xi;
-  dx_dzeta = dz_dzeta * cap_xi;
-  dy_dxi = dz_dxi * cap_eta;
-  dy_deta = (first_blending_factor +
-             second_blending_factor_over_rho_cubed * (1.0 + square(cap_xi))) *
-            cap_eta_deriv;
-  dy_dzeta = dz_dzeta * cap_eta;
-
-  // Implement Rotation:
-  if (direction_of_wedge_ == Direction<3>::lower_zeta()) {
-    dz_dxi *= -1;
-    dz_deta *= -1;
-    dz_dzeta *= -1;
-
-    dy_dxi *= -1;
-    dy_deta *= -1;
-    dy_dzeta *= -1;
-  }
-
-  if (direction_of_wedge_.axis() == Direction<3>::Axis::Eta) {
-    std::swap(dx_dxi, dy_dxi);
-    std::swap(dx_deta, dy_deta);
-    std::swap(dx_dzeta, dy_dzeta);
-    std::swap(dy_dxi, dz_dxi);
-    std::swap(dy_deta, dz_deta);
-    std::swap(dy_dzeta, dz_dzeta);
-
-    if (direction_of_wedge_.side() == Side::Lower) {
-      dy_dxi *= -1;
-      dy_deta *= -1;
-      dy_dzeta *= -1;
-
-      dz_dxi *= -1;
-      dz_deta *= -1;
-      dz_dzeta *= -1;
-    }
-  }
-
-  if (direction_of_wedge_.axis() == Direction<3>::Axis::Xi) {
-    std::swap(dx_dxi, dz_dxi);
-    std::swap(dx_deta, dz_deta);
-    std::swap(dx_dzeta, dz_dzeta);
-    std::swap(dz_dxi, dy_dxi);
-    std::swap(dz_deta, dy_deta);
-    std::swap(dz_dzeta, dy_dzeta);
-
-    if (direction_of_wedge_.side() == Side::Lower) {
-      dx_dxi *= -1;
-      dx_deta *= -1;
-      dx_dzeta *= -1;
-
-      dy_dxi *= -1;
-      dy_deta *= -1;
-      dy_dzeta *= -1;
-    }
-  }
-
+  ReturnType dz_dxi =
```

Comment:
Here is an implementation that is I think just as legible and uses 3 allocations instead of 9. I tested this on Gabe's branch.

```cpp
  ReturnType dz_dxi =
      -second_blending_factor_over_rho_cubed * cap_xi * cap_xi_deriv;
  ReturnType dx_dxi =
      (first_blending_factor +
       second_blending_factor_over_rho_cubed * (1.0 + square(cap_eta))) *
      cap_xi_deriv;
  ReturnType dy_dxi = dz_dxi * cap_eta;

  std::array<ReturnType, 3> dX_dlogical = discrete_rotation(
      orientation_of_wedge_,
      std::array<ReturnType, 3>{
          {std::move(dx_dxi), std::move(dy_dxi), std::move(dz_dxi)}});

  get<0, 0>(jacobian_matrix) = dX_dlogical[0];
  get<1, 0>(jacobian_matrix) = dX_dlogical[1];
  get<2, 0>(jacobian_matrix) = dX_dlogical[2];

  // reuse allocation, compute next subset of Jacobian
  dX_dlogical[2] =  // dz_deta
      -second_blending_factor_over_rho_cubed * cap_eta * cap_eta_deriv;
  dX_dlogical[0] =  // dx_deta
      dX_dlogical[2] * cap_xi;
  dX_dlogical[1] =  // dy_deta
      (first_blending_factor +
       second_blending_factor_over_rho_cubed * (1.0 + square(cap_xi))) *
      cap_eta_deriv;

  dX_dlogical =
      discrete_rotation(orientation_of_wedge_, std::move(dX_dlogical));

  get<0, 1>(jacobian_matrix) = dX_dlogical[0];
  get<1, 1>(jacobian_matrix) = dX_dlogical[1];
  get<2, 1>(jacobian_matrix) = dX_dlogical[2];

  dX_dlogical[2] =  // dz_dzeta
      first_blending_rate + second_blending_rate_over_rho;
  dX_dlogical[0] = dX_dlogical[2] * cap_xi;   // dx_dzeta
  dX_dlogical[1] = dX_dlogical[2] * cap_eta;  // dy_dzeta

  dX_dlogical =
      discrete_rotation(orientation_of_wedge_, std::move(dX_dlogical));

  get<0, 2>(jacobian_matrix) = dX_dlogical[0];
  get<1, 2>(jacobian_matrix) = dX_dlogical[1];
  get<2, 2>(jacobian_matrix) = dX_dlogical[2];
```